release.py

```diff
@@ -0,0 +1,139 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Release new version of artman.
+
+This script updates SHA hashes of latest commits for both gapic-generator
+and googleapis, and updates version number for subsequent (auto-)release.
+
+Usage:
+$ python3 release.py googleapis [sha]      # update googleapis commit hash
+$ python3 release.py gapic-generator [sha] # update gapic-generator commit hash
+$ python3 release.py 1.12.4                # update artman version
+"""
+
+import os.path
+import re
+import requests
+import sys
+from ruamel.yaml import YAML
+
+dockerfile_googleapis_regex = r'^(ENV GOOGLEAPIS_HASH )\S+()'
+dockerfile_gapic_generator_regex = r'^(ENV GAPIC_GENERATOR_HASH )\S+()'
+dockerfile_version_regex = r'^(ENV ARTMAN_VERSION )\S+()'
+setup_py_version_regex = r"^(current_version = ')\S+(')"
+
+github_gapic_generator_project = 'googleapis/gapic-generator'
+github_googleapis_project = 'googleapis/googleapis'
+
+
+def usage():
+    print('Usage:')
+    print(
+        '$ python3 release.py googleapis [sha]      # update googleapis commit hash'
+    )
+    print(
+        '$ python3 release.py gapic-generator [sha] # update gapic-generator commit hash'
+    )
+    print('$ python3 release.py 1.12.4                # update artman version')
+
+
+def get_latest_commit_sha(project):
+    url = 'https://api.github.com/repos/%s/branches/master' % project
+    # try to get GitHub credentials from config
+    auth = None
+    artman_config = os.path.join(os.environ['HOME'], '.artman', 'config.yaml')
+    try:
+        yaml = YAML()
+        with open(artman_config) as config_yaml:
+            config = yaml.load(config_yaml)
+            github_username = config['github']['username']
+            github_token = config['github']['token']
+            auth = requests.auth.HTTPBasicAuth(github_username, github_token)
+    except:
+        print('Cannot get GitHub token from %s, trying anonymously...' %
+              artman_config)
+
+    sha = None
+    try:
+        response = requests.get(url, auth=auth)
+        repo = response.json()
+        sha = repo['commit']['sha']
+    except:
+        print('Cannot get latest commit SHA. GitHub rate limit?')
+
+    return sha
+
+
+def update_file(filename, regex, version):
+    lines = []
+    with open(filename) as f:
+        for line in f:
+            line = re.sub(regex, r'\g<1>' + version + r'\g<2>', line)
+            lines.append(line)
+    with open(filename, 'w') as f:
+        f.write(''.join(lines))
+
+
+def main(argv):
+    if len(argv) == 1:
+        usage()
+        return 1
+
+    abspath = os.path.abspath(argv[0])
+    root_dir = os.path.dirname(abspath)
+    dockerfile = os.path.join(root_dir, 'Dockerfile')
+    setup_py = os.path.join(root_dir, 'setup.py')
+
+    sha = None
+    if len(argv) > 2:
```

Comment:
Consider using argparser for command-line args - take a look at the Python version bumping script in google-cloud-java: https://github.com/GoogleCloudPlatform/google-cloud-java/blob/master/utilities/bump_versions.py#L117